File: sample-java/src/main/java/com/android/mvvmarchitecturejavasample/entity/BaseEntity.java
package com.android.mvvmarchitecturejavasample.entity;

import com.imyyq.mvvm.base.IBaseResponse;

import org.jetbrains.annotations.Nullable;

public class BaseEntity<T> implements IBaseResponse<T> {

    public int errorCode;
    public String errorMsg;
    public T data;

    @Nullable
    @Override
    public Integer code() {
        return errorCode;
    }

    @Nullable
    @Override
    public String msg() {
        return errorMsg;
    }

    @Nullable
    @Override
    public T data() {
        return data;
    }

    @Override
    public boolean isSuccess() {
        return errorCode == 0;
    }
}

File: sample/src/main/java/com/imyyq/sample/entity/BaseEntity.kt
package com.imyyq.sample.entity

import com.imyyq.mvvm.base.IBaseResponse

data class BaseEntity<T>(
    var data: T?,
    var errorCode: Int?,
    var errorMsg: String?
) : IBaseResponse<T> {
    override fun code() = errorCode

    override fun msg() = errorMsg

    override fun data() = data

    override fun isSuccess() = errorCode == 0
}